lib/lambda_ethereum_consensus/state_transition/misc.ex

```diff
@@ -12,6 +13,69 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
     div(slot, slots_per_epoch)
   end
 
+  @doc """
+     Return the shuffled index corresponding to ``seed`` (and ``index_count``).
+  """
+  @spec compute_shuffled_index(SszTypes.uint64(), SszTypes.uint64(), SszTypes.bytes32()) ::
+          {:ok, SszTypes.uint64()} | {:error, String.t()}
+  def compute_shuffled_index(index, index_count, seed) do
+    result =
+      unless index < index_count and index_count > 0 do
+        {:error, "index not less than index count"}
+      else
+        shuffle_round_count = ChainSpec.get("SHUFFLE_ROUND_COUNT")
+
+        new_index =
+          Enum.reduce(0..(shuffle_round_count - 1), index, fn round, current_index ->
+            round_as_bytes =
+              <<round>> |> :binary.decode_unsigned() |> :binary.encode_unsigned(:little)
+
+            seed_as_bytes = seed |> :binary.decode_unsigned() |> :binary.encode_unsigned(:little)
```

Comment:
The `seed` here is already a binary, so no need to reinterpret it.

Reply:
Since, the hash function takes in binary data and the hash of value in little endian and big endian differs, doesn't it matter in this case?

Reply:
Still a bit unclear but I will read more on it. Thanks.